features/src/main/feature/feature.xml

```diff
@@ -84,6 +85,9 @@
     <feature name="openhab-ui-habot" description="HABot" version="${project.version}">
         <feature>openhab-ui-dashboard</feature>
         <bundle start-level="80">mvn:org.openhab.ui/org.openhab.ui.habot/${project.version}</bundle>
+        <bundle start-level="80">mvn:org.bouncycastle/bcprov-jdk15on/1.51</bundle>
+        <bundle start-level="80">mvn:org.apache.opennlp/opennlp-tools/1.8.3</bundle>
```

Comment:
@maggu2810 they all have a manifest, should this then be enough. And is the version number always repeated compared to the pom, or do we miss the last mile here?

Does openHAB now know where to find those jar's? Or do we need to push it to some deps repo, how would that work?

Reply:
I don't know how openHAB handles it features, but you do not need to add `start-level="80"` it should be the default and added by the karaf Maven plugin.

The feature is for runtime resolution, the GAVs does not need to fit to the ones in the POM.

We should relax the requirements a little bit and add `dependency="true"` for third party if possible. So, you do add the repective bundle / feature only if the capabilities are not satifsfied already.
E.g. if bcprov-jdk15on v1.60 is already installed, there is no need to install v1.51.

But you need to have a look at if the dependency is installed at all. It is only installed if someone requires it. If there is a import package or a require capability this will be done automatically.
If it is just to provide some logic, you need to add a requirement declaration yourself (you should find exampled in the tp feature).

Reply:
> But you need to have a look at if the dependency is installed at all. It is only installed if someone requires it. If there is a import package or a require capability this will be done automatically.

So you say that is long as its in the (generated) manifest that I don't have to put in the feature file, or do I misunderstand things now completely :-) Sorry, my OSGI experience is limited I'm more the Spring guy :-)

Reply:
No, you have to put it like here:

https://github.com/openhab/openhab-core/blob/45858f2/features/karaf/openhab-core/src/main/feature/feature.xml#L188-L189

(just that in your case you reference a bundle instead of a feature).